octue/resources/analysis.py

```diff
@@ -110,12 +110,23 @@ def finalise(self, output_dir=None):
         self.logger.debug("Validating serialised output json against twine")
         self.twine.validate(**serialised)
 
-        # Optionally write the serialised strand to disk
+        if output_dir is None:
+            return serialised
+
+        # Optionally write the serialised strand to disk and/or Google Cloud storage.
         for k in OUTPUT_STRANDS:
-            if output_dir and serialised[k] is not None:
+
+            if serialised[k] is not None:
                 file_name = get_file_name_from_strand(k, output_dir)
-                self.logger.debug(f"Writing {k} to file {file_name}")
-                with open(file_name, "w") as fp:
-                    fp.write(serialised[k])
+
+                if save_locally:
+                    with open(file_name, "w") as fp:
+                        fp.write(serialised[k])
+                    self.logger.debug("Wrote %r to file %r", k, file_name)
+
+                if upload_to_cloud:
+                    GoogleCloudStorageClient(project_name=project_name).upload_from_string(
```

Comment:
Not now but lets think about instantiating this at the start of the analysis then reusing the same client throught as needed.